### PR TITLE
Add JuMP interface

### PR DESCRIPTION
**Pull request by jd-foster**
This expands the options for trying different solvers.

### CODE REVIEW

File: src/jump.jl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    A, lb, ub = add_cycle_constraints_to_formulation(O)
+    JuMP_loadproblem!(model, A, O.l, O.u, O.c, lb, ub)
+
+    if !isempty(initial_solution) && use_warmstart
+        JuMP.set_start_value.(model[:x], initial_solution)
+    end
+    JuMP.optimize!(model)
+
+    if !JuMP.is_solved_and_feasible(model)
+        error("Problem not solve and feasible.")
```
Comment (jd-foster) [inline src/jump.jl:24]:
```suggestion
        error("Problem not solved or feasible.")
```

**Comment by jd-foster** [comment]
Usage example in external script:
```julia
using FeedbackArcSets
using HiGHS

bb = baharev_benchmark(100)
graph = bb.graph

##
resH = find_feedback_arc_set(graph;
    solver="jump",
    optimizer=HiGHS.Optimizer,
    allowableGap = 0.01, 
    log_level = 2
)
```

**Comment by jd-foster** [comment]
Definitely not necessary to have any specific tolerance, just an example of usage. This was replicating your HiGHS code, where you set this option to the MIP relative gap.

**Comment by GunnarFarneback** [comment]
Right, now I remember. It defaults to something non-zero, which didn't work so well, so I'm setting it explicitly to zero. An absolute gap could be used but you would need to reduce it gradually. I should have left a comment about it in the code.

**Comment by jd-foster** [comment]
Ok, I've introduced a `solver_options` keyword for passing in various settings, including the solver (the `optimizer`) that JuMP will call to solve the IP.
Usage:
```
resH = find_feedback_arc_set(graph;
    solver="jump",
    solver_options =[
        "optimizer" => HiGHS.Optimizer,
        "allowableGap" => 0.1, 
    ],
    log_level = 1
)
```

**Comment by jd-foster** [comment]
Checking again, I mis-remembered: `allowableGap` corresponds to the absolute gap e.g. `mip_abs_gap` in your original code.

File: src/highs.jl [hunk 2]
```diff
@@ -5,15 +5,16 @@ using HiGHS: Highs_create, Highs_setStringOptionValue, Highs_setSolution,
              Highs_getModelStatus, HighsInt
 
 function solve_IP(::Val{:highs}, O::OptProblem, initial_solution = Int[],
-                  use_warmstart = true; options...)
+                  use_warmstart = true; solver_options, options...)
```
Comment (GunnarFarneback) [inline src/highs.jl:8]: The `options` argument looks unused now.

File: src/highs.jl [hunk 3]
```diff
@@ -5,15 +5,16 @@ using HiGHS: Highs_create, Highs_setStringOptionValue, Highs_setSolution,
...
     model = Highs_create()
-    Highs_setStringOptionValue(model, "mip_rel_gap", "0")
```
Comment (GunnarFarneback) [inline src/highs.jl:10]: The gaps are somewhat problematic. They won't cause incorrect results but may interfere with convergence to optimality. Now that's by design what they are for, but should probably rather be specified for the minimal feedback arc set problem itself and then be converted to solver gaps. It's not clear how to do that though, so I'll have to think about it.

File: src/highs.jl (hunk 2 above)
Comment (jd-foster) [reply src/highs.jl:8]: The only instance now is in the`cbc.jl` code, but I can remove that too.

File: src/highs.jl (hunk 3 above)
Comment (jd-foster) [reply src/highs.jl:10]: I think this is best left as a solver option for the user with some default that works well for the problem generally.

File: src/highs.jl (hunk 2 above)
Comment (GunnarFarneback) [reply src/highs.jl:8]: Right, I can line that up.

**Comment by GunnarFarneback** [comment]
This looks good enough, so I'll merge it and do possible followups later.

Thanks for adding this. I've long wanted to have a JuMP option (for LongestPaths before this package) but never got around to learning how it should look. (I did try to use MOI when MPB was phased out, but that wasn't for the faint of heart.)

**Comment by GunnarFarneback** [comment]
Followup in #3.

**Pull request merged**